### PR TITLE
Add createLocal function

### PR DESCRIPTION
**Pull request by npty**
This PR add `createLocal` function in `networkUtils.ts` with options to specify `port`, `relayInterval`, `chains`, and `chainOutputPath`

### CODE REVIEW

File: src/networkUtils.ts [hunk 1]
```diff
@@ -248,14 +248,7 @@ function listen(port: number, callback: (() => void) | undefined = undefined) {
...
  */
-async function createNetwork(options: NetworkOptions = {
-    dbPath: undefined,
-    ganacheOptions: undefined,
-    port: undefined,
-    name: undefined,
-    chainId: undefined,
-    seed: undefined,
-}) {
+async function createNetwork(options: NetworkOptions) {
```
Comment (Foivos) [inline src/networkUtils.ts:251]: A bunch of the examples use no options so this change would break them.

File: src/networkUtils.ts [hunk 2]
```diff
@@ -447,9 +439,48 @@ function getDepositAddress(from: Network|string, to: Network|string, destination
...
+            to: user_address,
+            value: BigInt(100e18),
+        })).wait();
+        await (await user.sendTransaction({
+            to: deployer_address,
+            value: BigInt(100e18),
+        })).wait();
+        i++;
+    }
+    listen(8500);
```
Comment (Foivos) [inline src/networkUtils.ts:469]: use the port specified in options here.

File: src/networkUtils.ts [hunk 3]
```diff
@@ -447,9 +439,48 @@ function getDepositAddress(from: Network|string, to: Network|string, destination
     return address;
 }
 
+const chains = ["moonbeam", "avalanche", "fantom", "ethereum", "polygon"];
+const createLocal = async (userPrivateKey: string, deployerPrivateKey: string, options: CreateLocalOptions = {
```
Comment (Foivos) [inline src/networkUtils.ts:443]: this should be named something that describes what it does better, maybe "createAndExport"

File: src/networkUtils.ts [hunk 4]
```diff
@@ -447,9 +439,48 @@ function getDepositAddress(from: Network|string, to: Network|string, destination
     return address;
 }
 
+const chains = ["moonbeam", "avalanche", "fantom", "ethereum", "polygon"];
```
Comment (Foivos) [inline src/networkUtils.ts:442]: These should be a part of the params users can pass in.

File: src/networkUtils.ts (hunk 3 above)
Comment (Foivos) [reply src/networkUtils.ts:443]: userPrivateKey should be userAddress here, so that users can pass their metamask potentially.

**Review by Foivos — CHANGES_REQUESTED** [review]

**Comment by npty** [comment]
Changes as your comment. Please re-review 🙏🏻

File: src/Network.ts [hunk 5]
```diff
@@ -42,6 +42,7 @@ export interface NetworkOptions {
 
 export interface CreateLocalOptions {
   chainOutputPath: string,
+  chains: string[],
   relayInterval: number,
   port: number
 }
```
Comment (Foivos) [inline src/Network.ts:48]: add accountsToFund = []
add amount to fund = BigInt(10e18)

File: src/networkUtils.ts [hunk 6]
```diff
@@ -439,34 +439,32 @@ function getDepositAddress(from: Network|string, to: Network|string, destination
...
 }
 
-const chains = ["moonbeam", "avalanche", "fantom", "ethereum", "polygon"];
-const createLocal = async (userPrivateKey: string, deployerPrivateKey: string, options: CreateLocalOptions = {
+const createAndExport = async (userAddress: string, deployerAddress: string, options: CreateLocalOptions = {
   chainOutputPath: "./local.json",
+  chains: ["moonbeam", "avalanche", "fantom", "ethereum", "polygon"],
   port: 8500,
   relayInterval: 2000
 }) => {
```
Comment (Foivos) [inline src/networkUtils.ts:447]: add options specified above

File: src/networkUtils.ts [hunk 7]
```diff
@@ -439,34 +439,32 @@ function getDepositAddress(from: Network|string, to: Network|string, destination
...
+            to: userAddress,
             value: BigInt(100e18),
         })).wait();
         await (await user.sendTransaction({
-            to: deployer_address,
+            to: deployerAddress,
             value: BigInt(100e18),
         })).wait();
         i++;
     }
```
Comment (Foivos) [inline src/networkUtils.ts:466]: change this into a loop to fund all accounts specified.

**Review by Foivos — CHANGES_REQUESTED** [review]

**Comment by npty** [comment]
Fixedf

**Pull request merged**